apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,672 @@
+  pools = [] :: [{atom(), enabled | disabled | {pending, reference(), non_neg_integer()}}],
+  passive_pools = [] :: [atom()],
+  index = 0 :: non_neg_integer(),
+  priorities = #{} :: #{atom() => pos_integer()},
+  node_to_id = #{} :: #{atom() => iolist()},
+  check_ref :: reference()
+}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Category) ->
+  gen_fsm:start_link(?MODULE, [Category], []).
+
+run_task(Manager, #task{} = Task) ->
+  gen_fsm:send_event(Manager, {run, Task}).
+
+cancel_task(Manager, Id) ->
+  gen_fsm:send_event(Manager, {cancel, Id}).
+
+task_accepted(Manager, Ref, Node) ->
+  gen_fsm:send_event(Manager, {task_accepted, Ref, Node}).
+
+task_rejected(Manager, Ref, Node) ->
+  gen_fsm:send_event(Manager, {task_rejected, Ref, Node}).
+
+pool_connected(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_connected, Node}).
+
+pool_down(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_down, Node}).
+
+pool_enabled(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_enabled, Node}).
+
+pool_disabled(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_disabled, Node}).
+
+%% ------------------------------------------------------------------
+%% gen_fsm Function Definitions
+%% ------------------------------------------------------------------
+
+init([Category]) ->
+  self() ! start,
+  {ok, start, #state{category = Category}}.
+
+start(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = start_(Event, State),
+  erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}]).
+
+zero({check, Ref}, #state{check_ref = Ref} = State) ->
+  zero(check, State);
+
+zero({check, _}, State) ->
+  {next_state, zero, State};
+
+zero(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = zero_(Event, State),
+  case NextState of
+    zero ->
+      {next_state, NextState, NewState};
+    NextState ->
+      erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])
+  end.
+
+pools({check, Ref}, #state{check_ref = Ref} = State) ->
+  pools(check, State);
+
+pools({check, _}, State) ->
+  {next_state, pools, State};
+
+pools(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = pools_(Event, State),
+  case NextState of
+    pools ->
+      {next_state, NextState, NewState};
+    NextState ->
+      erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])
+  end.
+
+tasks({check, Ref}, #state{check_ref = Ref} = State) ->
+  tasks(check, State);
+
+tasks({check, _}, State) ->
+  {next_state, tasks, State};
+
+tasks(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = tasks_(Event, State),
+  case NextState of
+    tasks ->
+      {next_state, NextState, NewState};
+    NextState ->
+      erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])
```

Comment:
Line 135 is too long:       erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}]).